dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FirstOrderFilter.mo

```diff
@@ -0,0 +1,75 @@
+within Dynawo.NonElectrical.Blocks.Continuous;
+
+/*
+* Copyright (c) 2022, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block FirstOrderFilter "First order filter with absolute and rate limits, and a freezing flag"
+  import Modelica;
+
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Types.PerUnit DuMax "Maximun ramp rate";
+  parameter Types.PerUnit DuMin(start = -DuMax) "Minimun ramp rate";
+  parameter Types.Time tI(start = 1) "Filter time constant in s";
```

Comment:
What about the case where tI = 0 ? Should we have another model for that too ?

Reply:
I made another model.